crates/ibc/src/core/ics26_routing/handler.rs

```diff
@@ -53,12 +51,15 @@ pub fn decode(message: Any) -> Result<MsgEnvelope, RouterError> {
 /// and events produced after processing the input `msg`.
 /// If this method returns an error, the runtime is expected to rollback all state modifications to
 /// the `Ctx` caused by all messages from the transaction that this `msg` is a part of.
-pub fn dispatch<Ctx>(ctx: &mut Ctx, msg: MsgEnvelope) -> Result<HandlerOutput<()>, RouterError>
+pub(crate) fn dispatch<Ctx>(
```

Comment:
Similarly, let's keep this one `pub` (see [basecoin](https://github.com/informalsystems/basecoin-rs/blob/c5744f4a1eac9a63ef481410e52d9fb40363b97e/src/app/modules/ibc.rs#L301))

Reply:
Cancel that, I'll post a new solution soon